### PR TITLE
Metadata extraction updates

### PR DESCRIPTION
**Pull request by baitsguy**
1. Added a new notebook to extract metadata from ntsb samples
2. Handle inconsistent json responses from openai and guidance
3. Ability to customize max number of properties in schema

### CODE REVIEW

File: sycamore/utils/extract_json.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-        raise ValueError("JSON block not found in LLM response")
+    try:
+        return json.loads(payload)
+    except (ValueError, TypeError, JSONDecodeError):
+        pattern = r"```json([\s\S]*?)```"
+        match = re.match(pattern, payload)
+        if match:
+            return json.loads(match.group(1))
+        else:
+            raise ValueError("JSON block not found in LLM response: " + str(payload))
```
Comment (HenryL27) [inline sycamore/utils/extract_json.py:15]: What's the difference between these (besides the better error message)?

**Review by HenryL27 — APPROVED** [review]
LGTM, just one question

**Review by jonfritz — APPROVED** [review]
LGTM.

File: sycamore/utils/extract_json.py (hunk 1 above)
Comment (baitsguy) [reply sycamore/utils/extract_json.py:15]: we try to first parse the string as json directly, if that fails then we try to parse it as:
```json
{actual json}
```
which is what openai/guidance seems to be returning sometimes

**Pull request merged**